Makefile

```diff
@@ -821,6 +813,56 @@ ifneq ($(word 2,$(MAKECMDGOALS)),)
 export ENABLE_MSG_EXTRA := yes
 endif
 
+uavobjects_armsoftfp: TARGET=uavobjects_armsoftfp
+uavobjects_armsoftfp: OUTDIR=$(BUILD_DIR)/$(TARGET)
+uavobjects_armsoftfp: uavobjects
+	$(V1) mkdir -p $(OUTDIR)/dep
+	$(V1) cd $(ROOT_DIR)/flight/uavobjectlib && \
+		$(MAKE) -r --no-print-directory \
+		TCHAIN_PREFIX="$(ARM_SDK_PREFIX)" \
+		REMOVE_CMD="$(RM)" \
+		\
+		MAKE_INC_DIR=$(MAKE_INC_DIR) \
+		ROOT_DIR=$(ROOT_DIR) \
+		TARGET=$(TARGET) \
+		OUTDIR=$(OUTDIR) \
+		\
+		PIOS=$(PIOS) \
```

Comment:
Agreed, this sort of list is hard to keep up to date for similar reasons as pruning the set of unneeded ```#include``` lines from source files.  There's no explicit failure if you allow something that is no longer needed.  Or even if you allow something that is not needed and subsequently becomes wrong.

I agree that the submakefiles have a lot of unnecessary boilerplate lines that have been cut/pasted over time.  Getting that cleaned up would be a win for sure.

I can get behind nearly any approach that still provides containment/scoping for the makefiles in the targets dir.  If that ends up being something like exporting some top-level dirs to a common makefile that builds the scoped definitions and then unsets the top-level exports, that sounds good to me.  I'm just a bit wary of going back to the bad-old-days where these top-level dirs were globals throughout the build.

The overall intention of having some sort of API/scoping for the submakefiles is the thing I'm hoping to preserve.  I think there are a lot of different ways to implement it and there's lots of room for moving to new/better ways in another PR.

In this PR, the export of the BUILD_DIR and extra parameters in the new lib build target are the only sticking points for me to merge.  Like I said before though, I'll still defer to a 2nd reviewer on both points if that's what they recommend.